### PR TITLE
Publish check added

### LINKED ISSUE #3: Check time constraints on pages
LUYA CMS allows to define time range for when a page is public, this is currently not reflected when adding URLs to the sitemap.

### PR DESCRIPTION
**Pull request by SOHELAHMED7**
fix #3

### CODE REVIEW

File: src/controllers/SitemapController.php [hunk 1]
```diff
@@ -152,4 +142,19 @@ private function getRelativeUriByNavItem($navItem, $ignoreNavIds)
...
+     */
+    private function getBasicQuery()
+    {
+        return Nav::find()->where([
+                'is_deleted' => false,
+                'is_offline' => false,
+                'is_draft' => false
+            ])
+            ->andWhere(['or', ['publish_from' => null], ['<=', 'publish_from', new Expression('UNIX_TIMESTAMP()')]])
+            ->andWhere(['or', ['publish_till' => null], ['>=', 'publish_till', new Expression('UNIX_TIMESTAMP()')]]);
```
Comment (cebe) [inline src/controllers/SitemapController.php]: you can use `time()` instead.

File: src/controllers/SitemapController.php [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 namespace cebe\luya\sitemap\controllers;
 
+use yii\db\Expression;
```
Comment (cebe) [inline src/controllers/SitemapController.php]: unnecessary import.

File: src/controllers/SitemapController.php [hunk 3]
```diff
@@ -152,4 +142,19 @@ private function getRelativeUriByNavItem($navItem, $ignoreNavIds)
 
         return $fullUriPath;
     }
+
+    /**
+     * Common query building part
+     * @return \yii\db\ActiveQuery
+     */
+    private function getBasicQuery()
```
Comment (cebe) [inline src/controllers/SitemapController.php]: better name: getBaseQuery()

File: src/controllers/SitemapController.php (hunk 2 above)
Comment (SOHELAHMED7) [reply src/controllers/SitemapController.php]: fixed by a262fd41734f41d6d099be46e9814fd1359d1ffe

File: src/controllers/SitemapController.php (hunk 3 above)
Comment (SOHELAHMED7) [reply src/controllers/SitemapController.php]: fixed by a262fd41734f41d6d099be46e9814fd1359d1ffe

**Review by cebe — APPROVED** [review]

**Comment by cebe** [comment]
Merged, thank you!

**Pull request merged**